src/adapters/audienceNetwork.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * @file AudienceNetwork adapter.
+ */
+import { ajax } from '../ajax';
+import { createBid } from '../bidfactory';
+import { addBidResponse } from '../bidmanager';
+import { STATUS } from '../constants.json';
+import { format } from '../url';
+import { logError } from '../utils';
+import { createNew } from './adapter';
+
+const baseAdapter = createNew('audienceNetwork');
+const setBidderCode = baseAdapter.setBidderCode;
+const getBidderCode = baseAdapter.getBidderCode;
+
+/**
+ * Does this bid request contain valid parameters?
+ * @param {Object} bid
+ * @returns {Boolean}
+ */
+const validateBidRequest = bid =>
+  typeof bid.params === 'object' &&
+  typeof bid.params.placementId === 'string' &&
+  bid.params.placementId.length > 0 &&
+  Array.isArray(bid.sizes) && bid.sizes.length > 0;
+
+/**
+ * Does this bid request contain valid sizes?
+ * @param {Object} bid
+ * @returns {Boolean}
+ */
+const validateBidRequestSizes = bid => {
+  bid.sizes = bid.sizes.map(flattenSize);
+  return bid.sizes.every( size =>
+    ['native', 'fullwidth', '300x250', '320x50'].includes(size) );
+};
+
+/**
+ * Flattens a 2-element [W, H] array as a 'WxH' string,
+ * otherwise passes value through.
+ * @params {Array|String} size
+ * @returns {String}
+ */
+const flattenSize = size =>
+  (Array.isArray(size) && size.length === 2) ? `${size[0]}x${size[1]}` : size;
+
+/**
+ * Does the search part of the URL contain "anhb_testmode"
+ * and therefore indicate testmode should be used?
+ * @returns {String} "true" or "false"
+ */
+const isTestmode = () => Boolean(
```

Comment:
This detects the addition of `?anhb_testmode` in a given page's URL so only test bids are returned.  Those test bids won't be honoured beyond the initial response. I guess/hope any official FB docs will include mention of this but happy to add this to the "quick start" in the example too if required.